File: app/src/main/java/knf/kuma/tv/TVBaseActivity.java
package knf.kuma.tv;


import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentTransaction;
import knf.kuma.R;
import knf.kuma.commons.SSLSkipper;

public class TVBaseActivity extends FragmentActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.tv_activity_main);
        SSLSkipper.skip();
    }

    public void addFragment(Fragment fragment) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.tv_frame_content, fragment);
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/knf/kuma/tv/TVBaseActivity.kt
package knf.kuma.tv


import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import knf.kuma.R
import knf.kuma.backup.firestore.FirestoreManager
import knf.kuma.commons.SSLSkipper
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlin.contracts.ExperimentalContracts

@ExperimentalCoroutinesApi
@ExperimentalContracts
open class TVBaseActivity : FragmentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.tv_activity_main)
        SSLSkipper.skip()
        FirestoreManager.start()
    }

    fun addFragment(fragment: Fragment) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.tv_frame_content, fragment)
        fragmentTransaction.commit()
    }
}
